### PR TITLE
feat: adds support for BIP39 mnemonics

### PR DESCRIPTION
**Pull request by SweeXordious**
## Overview

Closes https://github.com/celestiaorg/orchestrator-relayer/issues/495

## Checklist

- [ ] New and updated code has appropriate documentation
- [ ] New and updated code has new and/or updated testing
- [ ] Required CI checks are passing
- [ ] Visual proof for any user facing features like CLI or documentation updates
- [ ] Linked issues closed with keywords

## Summary by CodeRabbit

- **New Features**
  - Added the ability to derive a private key from a mnemonic phrase.
  - Introduced the feature to import an EVM address from a 24-word BIP39 mnemonic phrase.
- **Tests**
  - Added tests to validate the functionality of deriving a private key from a mnemonic phrase.

### CODE REVIEW

File: cmd/blobstream/keys/evm/evm.go [hunk 1]
```diff
@@ -520,9 +628,35 @@ func GetNewPassphrase() (string, error) {
...
 		if bytes.Equal(bzPassphrase, bzPassphraseConfirm) {
+			fmt.Println()
 			break
 		}
 		fmt.Print("\npassphrase and confirmation mismatch.\n")
 	}
 	return string(bzPassphrase), nil
 }
+
+// MnemonicToPrivateKey derives a private key from the provided mnemonic
```
Comment (SweeXordious) [inline cmd/blobstream/keys/evm/evm.go:639]:
```suggestion
// MnemonicToPrivateKey derives a private key from the provided mnemonic.
```

File: cmd/blobstream/keys/evm/evm.go [hunk 2]
```diff
@@ -520,9 +628,35 @@ func GetNewPassphrase() (string, error) {
...
+// It uses the geth.LegacyLedgerBaseDerivationPath, i.e. m/44'/60'/0'/0, to generate
```
Comment (SweeXordious) [inline cmd/blobstream/keys/evm/evm.go:640]:
```suggestion
// It uses the Ledger derivation path, geth.LegacyLedgerBaseDerivationPath, i.e. m/44'/60'/0'/0, to generate
```

**Review by SweeXordious — COMMENTED** [review]
[note for reviewers]
It would be helpful to test the command while reviewing to see if the prompt messages are correct and intuitive.

File: cmd/blobstream/keys/evm/evm.go [hunk 3]
```diff
@@ -379,6 +426,87 @@ func ImportECDSA(serviceName string) *cobra.Command {
 	return keysConfigFlags(&cmd, serviceName)
 }
 
+func ImportMnemonic(serviceName string) *cobra.Command {
+	cmd := cobra.Command{
+		Use:   "mnemonic",
+		Args:  cobra.ExactArgs(0),
+		Short: "import an EVM address from a 24 words BIP39 mnemonic phrase",
```
Comment (evan-forbes) [inline cmd/blobstream/keys/evm/evm.go:433]: if the key is being used to sign something and not just an address, then I think this terminology makes more sense 
```suggestion
		Short: "import a EVM private key from a 24 words BIP39 mnemonic phrase",
```

**Review by evan-forbes — APPROVED** [review]

**Review by evan-forbes — APPROVED** [review]
:+1:

**Pull request merged**